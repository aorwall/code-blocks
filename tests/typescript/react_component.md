# Chunk 1
```tsx
import React, { useState } from 'react';

// Define a type for the props
interface MyComponentProps {
  initialCount?: number;
}

// Define the component
// ...
```

# Chunk 2
```tsx


// ...
const MyComponent: React.FC<MyComponentProps> = ({ initialCount = 0 }) => {
  // Use a state variable
  const [count, setCount] = useState(initialCount);

  const resetCount = () => {
    setCount(initialCount);
  };

  return (
    <div>
      <p>You clicked {count} times</p>
      <button onClick={() => setCount(count + 1)}>
        Click me
      </button>
    </div>
  );
}

export default MyComponent;

```

