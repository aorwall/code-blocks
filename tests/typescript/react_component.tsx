import React, { useState } from 'react';

// Define a type for the props
interface MyComponentProps {
  initialCount?: number;
}

// Define the component
const MyComponent: React.FC<MyComponentProps> = ({ initialCount = 0 }) => {
  // Use a state variable
  const [count, setCount] = useState(initialCount);

  return (
    <div>
      <p>You clicked {count} times</p>
      <button onClick={() => setCount(count + 1)}>
        Click me
      </button>
    </div>
  );
}

export default MyComponent;
