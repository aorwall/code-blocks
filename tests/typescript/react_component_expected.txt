 0 module ``
  1 import `import React, { useState } from 'react';`
  1 comment `// Define a type for the props`
  1 class `interface MyComponentProps {\n  initialCount?: number;\n}`
  1 comment `// Define the component`
  1 code `const MyComponent: React.FC<MyComponentProps>`
   2 code `=`
   2 code `({ initialCount = 0 }) =>`
    3 block_delimiter `{`
    3 comment `// Use a state variable`
    3 code `const [count, setCount]`
     4 code `=`
     4 code `useState(initialCount)`
     4 code `;`
    3 code `const resetCount`
     4 code `=`
     4 code `() =>`
      5 block_delimiter `{`
      5 code `setCount(initialCount);`
      5 block_delimiter `}`
     4 code `;`
    3 statement `return`
     4 code ``
      5 block_delimiter `(`
      5 code `<div>`
       6 code ``
       6 code `<p>`
        7 code `You clicked`
        7 code `{count}`
        7 code `times`
        7 code `</p>`
       6 code ``
       6 code `<button onClick={() => setCount(count + 1)}>`
        7 code `Click me`
        7 code `</button>`
       6 code ``
       6 code `</div>`
      5 block_delimiter `)`
     4 code `;`
    3 block_delimiter `}`
  1 code `export default MyComponent;`
  1 space ``
